### ISSUE TITLE
AltiProfil 0.5.0 LM 3.8.0-rc.4 Position of the profile is missing in the graphic

### ISSUE THREAD

**Issue by guenterw** [arno974/lizmap-altiProfil#41]
The position of the profile is not shown in the graphic. The profile display appears as usual.
Installed under Ubuntu 22.04. and 24.04. with Lizmap 3.8.0-rc.4

![image](https://github.com/user-attachments/assets/21920a36-f9ad-40a9-9bb2-f864d6ec07f0)

https://cx21-8.webgis.biz/index.php/view/map?repository=zaltiprofil&project=test_profil_3_34#7.798018,48.214294,7.823238,48.226345|Politischegrenzen,ALKIS,Gebaeudeschraffuren,test_dgm_db,Luftbilder,DOP_2023,DOP_2021,2021|default,default,default,default,,,,|1,1,1,1,1,1,1,1

With Lizmap 3.7.10 and AltiProfil 0.4.2 it works perfectly on the servers.
Am I making a mistake with the installation or is it a bug?

**Comment by guenterw** [comment]
Even with AltiProfil 0.5.2 and Lizmap 3.8.2, the position of the profile is not displayed in the graphic:
![image](https://github.com/user-attachments/assets/e5b902a3-8147-4628-a5d6-0c30835702fa)

![image](https://github.com/user-attachments/assets/913eb9eb-6558-4681-9a76-0e399ce925af)

Am I doing something wrong or can someone confirm the error?

**Comment by Gustry** [comment]
@guenterw Just look before your message, there is an open PR which will close this ticket ;-)

**Comment by guenterw** [comment]
@Gustry, Thanks, but I didn't understand this ‘PR’ and didn't associate it with solving this problem.

Will this tool be developed further in the future?
I have customer requests for changes/additions (against payment), which I forwarded to Arnaud Vandecasteele by e-mail in June, but I have not received a reply.

**Comment by Gustry** [comment]
> Thanks, but I didn't understand this ‘PR’ and didn't associate it with solving this problem.

In the PR description, it says `Fix #41` so according to @nworr merging this PR will indeed solve this issue.
If I understood the problem, it's because of the layer used by AltiProfil to show the geometry is **under** the main map coming from the QGIS projects. We need to bring the layer **up** in OpenLayers.

> Will this tool be developed further in the future?

Yes, this module is active, you can see the [list of commits](https://github.com/arno974/lizmap-altiProfil/commits/master/), and there are some devs. I'm not sure if @arno974 want to take the lead on your requests, I can check with him. You can also create feature request on GitHub and ask any company to work on an opensource product :)

**Comment by guenterw** [comment]
Thanks for your feedback and explanations

**Comment by arno974** [comment]
Hi @Gustry ,

They are two pending pull requests. If one of them resolve this issue we should accept the pull d'ont you think ?

**Comment by Gustry** [comment]
Yes, as said, #43 will fix this issue #41 but I guess @nworr is waiting a review or confirmation ;-)

**Comment by nworr** [comment]
Yes, #43 is waiting a confirmation.
 Using script override in firefox's developer tools, on @guenterw example, it works perfectly, but I dont undertand why some layers are now using using Zindex, i didn't find what introduces this behaviour in lizmap.

**Comment by arno974** [comment]
For me it seems all good. Your dev skills are far better than mine anyway :)

**Comment by nworr** [comment]
#43 updated, way much more elegant and efficient. release is soon

**Comment by guenterw** [comment]
@nworr , @Gustry Thank you for the adjustments / troubleshooting
The position of the profile is now visible again in the site plan. 
One small detail is still missing. If a certain position is marked in the profile, this position was previously also displayed in the site plan. It is currently not displayed in 0.5.3.

Lizmap 3.8.2 mit altiProfil 0.5.3 (https://cx21-8.webgis.biz)
![image](https://github.com/user-attachments/assets/b64329c9-55a7-4660-a244-62bec47a6e84)

Lizmap 3.7.11-pre (https://cx21-7.webgis.biz)
![image](https://github.com/user-attachments/assets/7c1a61d1-1487-4413-82e9-574693c3be07)

**Comment by Gustry** [comment]
Can you create a separate issue ? Thanks for testing.

**Comment by guenterw** [comment]
and another outlook on the 3.9.0-pre
The profile is displayed incorrectly, but the elevations are now nicely rounded to 2 decimal places (see green marker).
Can this rounding of heights also be realized for version 3.8.x ?

Lizmap 3.9.0-pre with altiProfil 0.5.3 (https://cx21-9.webgis.biz)
![image](https://github.com/user-attachments/assets/dcca64e4-7eac-446f-9b5b-b3767a59f183)

**Comment by guenterw** [comment]
@Gustry , ok

**Comment by Gustry** [comment]
> and another outlook on the 3.9.0-pre

Can you create another ticket for it @guenterw ? It's always better to create new, separate ticket for each issue, instead of on a closed ticket ;-) Thanks

**Comment by guenterw** [comment]
I wasn't sure whether this was interesting at this early stage of 3.9. But I'm happy to do it
